packages/agreement-process/src/services/agreementSuspensionProcessor.ts

```diff
@@ -124,9 +123,19 @@ export async function suspendAgreementLogic({
     ...updateSeed,
   };
 
-  return toCreateEventAgreementUpdated(
-    updatedAgreement,
-    agreement.metadata.version,
-    correlationId
-  );
+  if (authData.organizationId === agreement.data.consumerId) {
+    return toCreateEventAgreementSuspendedByConsumer(
+      updatedAgreement,
+      agreement.metadata.version,
+      correlationId
+    );
+  } else if (authData.organizationId === agreement.data.producerId) {
+    return toCreateEventAgreementSuspendedByProducer(
+      updatedAgreement,
+      agreement.metadata.version,
+      correlationId
+    );
+  } else {
+    throw new Error("Unexpected state");
```

Comment:
I know this should not happen, but just in case can you add a meaningful message?